go/vt/vtadmin/cluster/cluster.go

```diff
@@ -1026,6 +1030,148 @@ func (c *Cluster) GetSchema(ctx context.Context, keyspace string, opts GetSchema
 	return c.getSchemaFromTablets(ctx, keyspace, tabletsToQuery, opts)
 }
 
+// GetSchemas returns all of the schemas across all keyspaces in the cluster.
+func (c *Cluster) GetSchemas(ctx context.Context, opts GetSchemaOptions) ([]*vtadminpb.Schema, error) {
+	span, ctx := trace.NewSpan(ctx, "Cluster.GetSchemas")
+	defer span.Finish()
+
+	if opts.TableSizeOptions == nil {
+		opts.TableSizeOptions = &vtadminpb.GetSchemaTableSizeOptions{
+			AggregateSizes:          false,
+			IncludeNonServingShards: false,
+		}
+	}
+
+	if opts.BaseRequest == nil {
+		opts.BaseRequest = &vtctldatapb.GetSchemaRequest{}
+	}
+
+	if opts.TableSizeOptions.AggregateSizes && opts.BaseRequest.TableNamesOnly {
+		log.Warningf("GetSchemas(cluster = %s) size aggregation is incompatible with TableNamesOnly, ignoring the latter in favor of aggregating sizes", c.ID)
+		opts.BaseRequest.TableNamesOnly = false
+	}
+
+	AnnotateSpan(c, span)
+	annotateGetSchemaRequest(opts.BaseRequest, span)
+	vtadminproto.AnnotateSpanWithGetSchemaTableSizeOptions(opts.TableSizeOptions, span)
+
+	var (
+		m   sync.Mutex
+		wg  sync.WaitGroup
+		rec concurrency.AllErrorRecorder
+
+		tablets   []*vtadminpb.Tablet
+		keyspaces []*vtadminpb.Keyspace
+
+		schemas []*vtadminpb.Schema
+	)
+
+	// Start by collecting the tablets and keyspace names concurrently.
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+
+		var err error
+		tablets, err = c.GetTablets(ctx)
+		if err != nil {
+			rec.RecordError(err)
+			return
+		}
+	}()
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+
+		// TODO: (ajm188) we can't use c.GetKeyspaces because it also makes a
+		// FindAllShardsInKeyspace call for each keyspace, which we may or may
+		// not need. Refactor that method so we can get better code reuse.
```

Comment:
This comes from [here](https://github.com/vitessio/vitess/pull/10108/files#diff-3e212bf9fa780da22b2bbce441ebb9ed37a8c590bfa017c13ecb6e8b245fd5e3L847-L856) (which, fun fact, was _also_ buggy because it bypassed the topo read pool 😭 ... i have put so many bugs in here and i have .... Plans to clean this up, but I'll do that separately.